flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/validation/FlinkResourceValidator.java

```diff
@@ -17,18 +17,19 @@
 
 package org.apache.flink.kubernetes.operator.validation;
 
+import org.apache.flink.core.plugin.Plugin;
 import org.apache.flink.kubernetes.operator.crd.FlinkDeployment;
 import org.apache.flink.kubernetes.operator.crd.FlinkSessionJob;
 
 import java.util.Optional;
 
 /** Validator for different resources. */
-public interface FlinkResourceValidator {
+public interface FlinkResourceValidator extends Plugin {
```

Comment:
I am not sure whether we need to introduce the `FlinkResourceValidatorFactory`, which actually extends the `Plugin` interface. The advantage is we do not need a default constructor for `FlinkResourceValidator`.

Another thing is we might not need to extend `Plugin` interface since we do not call `#configure()`. If some users implement the `#configure()`, it could not work.

Reply:
@wangyang0918, I have added the call of the `configure()` in `ValidatorUtils`. IMO, the advantage isn't too necessary to  introduce the `FlinkResourceValidatorFactory`.